packages/black-forest-labs/src/bfl-image-model.ts

```diff
@@ -0,0 +1,280 @@
+        ...(providerOptions.bfl ?? {}),
+        ...(providerOptions['black-forest-labs'] ?? {}),
+      },
+      failedResponseHandler: bflFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(bflSubmitSchema),
+      abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    const pollUrl = submit.value.polling_url;
+    const requestId = submit.value.id;
+    const imageUrl = await this.pollForImageUrl({
+      pollUrl,
+      requestId,
+      abortSignal,
+    });
+
+    const { value: imageBytes, responseHeaders } = await getFromApi({
+      url: imageUrl,
+      abortSignal,
+      failedResponseHandler: createStatusCodeErrorResponseHandler(),
+      successfulResponseHandler: createBinaryResponseHandler(),
+      fetch: this.config.fetch,
+    });
+
+    return {
+      images: [imageBytes],
+      warnings,
+      response: {
+        modelId: this.modelId,
+        timestamp: currentDate,
+        headers: responseHeaders,
+      },
+    };
+  }
+
+  private async pollForImageUrl({
+    pollUrl,
+    requestId,
+    abortSignal,
+  }: {
+    pollUrl: string;
+    requestId: string;
+    abortSignal: AbortSignal | undefined;
+  }): Promise<string> {
+    const url = new URL(pollUrl);
+    if (!url.searchParams.has('id')) {
+      url.searchParams.set('id', requestId);
+    }
+
+    while (true) {
+      const response = await (this.config.fetch ?? fetch)(url.toString(), {
+        method: 'GET',
+        headers: Object.fromEntries(
+          Object.entries(combineHeaders(await resolve(this.config.headers))).filter(
+            ([, v]) => v !== undefined,
+          ),
+        ) as Record<string, string>,
+        signal: abortSignal,
+      });
+      if (!response.ok) {
+        const errorBody = await safeParseJson(response);
+        const message =
+          bflErrorToMessage(errorBody) ?? response.statusText ?? 'BFL error';
+        throw new Error(message);
+      }
+      const json = (await response.json()) as unknown;
+      let status: string | undefined;
+      let sample: string | undefined;
+      if (isJsonObject(json)) {
+        const statusVal = json['status'];
+        const stateVal = json['state'];
+        if (typeof statusVal === 'string') status = statusVal;
+        else if (typeof stateVal === 'string') status = stateVal;
+
+        const resultVal = json['result'];
+        if (isJsonObject(resultVal)) {
+          const sampleVal = resultVal['sample'];
+          if (typeof sampleVal === 'string') sample = sampleVal;
+        }
+      }
+      if (status === 'Ready') {
+        if (typeof sample === 'string') {
+          return sample;
+        }
+        throw new Error('BFL poll response is Ready but missing result.sample');
+      }
+      if (status === 'Error' || status === 'Failed') {
+        const msg =
+          bflErrorToMessage(json) ?? 'Black Forest Labs generation failed.';
+        throw new Error(msg);
+      }
+
+      await delay(
```

Comment:
The polling we did in `luma-image-model.ts` seems simpler, do we need a separate delay method like this?